src/components/application_manager/src/resumption/resumption_data_json.cc

```diff
@@ -152,32 +160,38 @@ void ResumptionDataJson::IncrementIgnOffCount() {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock autolock(resumption_lock_);
   Json::Value to_save = Json::arrayValue;
-  for (Json::Value::iterator it = GetSavedApplications().begin();
-       it != GetSavedApplications().end();
-       ++it) {
-    if ((*it).isMember(strings::ign_off_count)) {
-      Json::Value& ign_off_count = (*it)[strings::ign_off_count];
+  Json::Value dictionary;
+  resumption::LastStateAccessor accessor = last_state_wrapper_->get_accessor();
+  dictionary = accessor.GetData().dictionary();
+  Json::Value& saved_apps = GetSavedApplications(dictionary);
+  for (auto json_app = saved_apps.begin(); json_app != saved_apps.end();
+       ++json_app) {
+    if ((*json_app).isMember(strings::ign_off_count)) {
+      Json::Value& ign_off_count = (*json_app)[strings::ign_off_count];
       const uint32_t counter_value = ign_off_count.asUInt();
       ign_off_count = counter_value + 1;
     } else {
       LOG4CXX_WARN(logger_, "Unknown key among saved applications");
-      Json::Value& ign_off_count = (*it)[strings::ign_off_count];
+      Json::Value& ign_off_count = (*json_app)[strings::ign_off_count];
       ign_off_count = 1;
     }
-    to_save.append(*it);
+    to_save.append(*json_app);
   }
-  SetSavedApplication(to_save);
-  SetLastIgnOffTime(time(NULL));
-  LOG4CXX_DEBUG(logger_, GetResumptionData().toStyledString());
+  SetSavedApplication(to_save, dictionary);
+  SetLastIgnOffTime(time(NULL), dictionary);
+  accessor.GetMutableData().set_dictionary(dictionary);
+  LOG4CXX_DEBUG(logger_, GetResumptionData(dictionary).toStyledString());
 }
 
 void ResumptionDataJson::DecrementIgnOffCount() {
   using namespace app_mngr;
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock autolock(resumption_lock_);
 
-  for (Json::Value::iterator it = GetSavedApplications().begin();
-       it != GetSavedApplications().end();
+  resumption::LastStateAccessor accessor = last_state_wrapper_->get_accessor();
+  Json::Value dictionary = accessor.GetData().dictionary();
+  Json::Value& saved_apps = GetSavedApplications(dictionary);
+  for (Json::Value::iterator it = saved_apps.begin(); it != saved_apps.end();
```

Comment:
maybe range-based

Reply:
@SNiukalov fixed in https://github.com/smartdevicelink/sdl_core/pull/2626/commits/f4e32226bda388a572a9edc282e74636a66ad925